src/gui/uithememanager.cpp

```diff
@@ -55,16 +150,49 @@ void UIThemeManager::initInstance()
         m_instance = new UIThemeManager;
 }
 
-UIThemeManager::UIThemeManager()
+UIThemeManager::UIThemeManager() : m_useCustomStylesheet(false)
 {
     const Preferences *const pref = Preferences::instance();
-    if (pref->useCustomUITheme()
-        && !QResource::registerResource(pref->customUIThemePath(), "/uitheme"))
-        LogMsg(tr("Failed to load UI theme from file: \"%1\"").arg(pref->customUIThemePath()), Log::WARNING);
+    bool useCustomUITheme = pref->useCustomUITheme();
 
-#if (defined(Q_OS_UNIX) && !defined(Q_OS_MACOS))
+#ifdef QBT_SYSTEMICONS
     m_useSystemTheme = pref->useSystemIconTheme();
+    m_systemIconMap = systemIconConfig;
 #endif
+
+    QHash<QString, QString> customIconMap;
+
+    if (useCustomUITheme) {
+        try {
+            if (!QResource::registerResource(pref->customUIThemePath(), "/uitheme"))
+                throw ThemeError(tr("Failed to load UI theme from file: \"%1\".")
+                                 .arg(pref->customUIThemePath()));
+            m_useCustomStylesheet = QFile::exists(":uitheme/stylesheet.qss");
+            m_flagsDir = ":uitheme/icons/flags/";
+
+            // by chaining following functions, we ignore the erroed values in customIconMap
+            customIconMap = resolveDirs(loadIconConfig(":uitheme/icons/iconconfig.json"), ":uitheme/icons/");
```

Comment:
Maybe ask the custom theme for absolute icon path, this way they are not just limited to qrc but reference icon from somewhere else also.